CITATION.cff

```diff
@@ -0,0 +1,25 @@
+cff-version: 1.2.0
+message: "Please cite the following works when using this software."
+type: software
+title: "hist: v2.6.1"
```

Comment:
I'm of the opinion that including the version number in software citation is highly important, but this does mean that someone then needs to start keeping this synced or use a bump tool like `pyhf` does. I can understand that might not be desirable.